### DISCUSSION TITLE
Keep uploaded file after validation error

### DISCUSSION THREAD

**Discussion by mobielbekeken-jan** [Laravel-Backpack/community-forum#408] [Ideas]
When you for example create a monster in the backpack demo and you leave the 'text' field in the 'simple' tab empty, u will get a validation error about that. The value for for example 'textarea' will be left intact which is good, but if you have uploaded a file its gone and you won't notice that. The user is in the assumption everything is saved but everything is except the uploaded files.

Best solution would be to hold the old value for an upload field, next best solution would be to warn user that uploaded files are not stored because of the validation error.

**Comment by pxpm** [comment]
Hello @mobielbekeken-jan thanks for the suggestion. 

Indeed is something that bothered us for so long, that we are **already** working on a solution for that. 
There is no magic behind it, the fact that you cannot manually set the value of a file upload is a **security feature** not an issue. 

We are implementing an ajax upload solution that would allow us to solve that issue by storing a temporary file when user selects it. 

We expect to ship it by next month.

If you are in a hurry to have some kind of message on field when there are validation errors, you can create a copy of the upload field, let's say `php artisan backpack:field upload_with_validation_error --from=upload` (This will create a copy of the original upload file called `upload_with_validation_error.blade.php` in your app/resources folder). 

You can then manually add a message on that field if there are `old()` values in the session. 

Hope it helps,
Cheers

**Comment by mobielbekeken-jan** [comment]
Awesome! Thank you, looking forward to the ajax upload solution, temporary validation error will do for now :)
